core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -72,13 +75,15 @@ private[spark] abstract class Task[T](
    * Called by [[org.apache.spark.executor.Executor]] to run this task.
    *
    * @param taskAttemptId an identifier for this task attempt that is unique within a SparkContext.
-   * @param attemptNumber how many times this task has been attempted (0 for the first attempt)
+   * @param attemptNumber how many times this task has been attempted (0 for the first attempt).
+   * @param executorPlugins the plugins which will be notified of the run of this task.
    * @return the result of the task along with updates of Accumulators.
    */
   final def run(
       taskAttemptId: Long,
       attemptNumber: Int,
-      metricsSystem: MetricsSystem): T = {
+      metricsSystem: MetricsSystem,
+      executorPlugins: Seq[ExecutorPlugin]): T = {
```

Comment:
Can make this parameter optional
```suggestion
      executorPlugins: Seq[ExecutorPlugin] = Seq.empty): T = {
```